consistency/tool/clustering/clustering.py

```diff
@@ -43,16 +44,74 @@ def extract_indices_dice(x, y):
         data = np.arange(len(self.words)).reshape(-1, 1)
     
         try:
-            db = dbscan(data, metric=extract_indices_dice, eps=0.3, min_samples=2, algorithm='brute')
+            _, labels = dbscan(data, metric=extract_indices_dice, eps=0.3, min_samples=2, algorithm='brute')
         except Exception as e:
             print(e, " Word Clustering Failed!")
             return None
     
         #if all labels are -1, DBSCAN detected no possible clusters. 
-        if np.count_nonzero(db[1] == -1) == db[1].size:
+        if np.count_nonzero(labels == -1) == labels.size:
             warnings.warn("There were no clusters detected. All words are unique.")
     
-        return db[1]
+        return labels
+
+    def create_word_groups(self):
+        
+
+        labels = np.array(self.cluster())
+
+        temp_dict  = {}
+        if not labels.size or not self.words:
+            return None
+        for j in range(len(self.words)):
+            word_label = labels[j]
+    
+            if word_label in temp_dict:
+                temp_dict[word_label].append(self.words[j])
+                        
+            else:
+                temp_dict[word_label] = [self.words[j]]
+
+        word_groups = []
+
+        def find_clusterID(similar_list):
```

Comment:
Any reason to make this a nested function? I am not too familiar with coding in python but wanted to bring your attention to this -> go/pystyle#nested-classes-functions-decision.

Reply:
I see the style guide says it is fine to nest functions that are only used inside of a very limited scope. They say they are fine  with caveats. I am going to make it stay that way for consistency purposes (we have a nested function inside the clustering function, too.) unless you have strong opinions about them.